feature-1: UserActivityAnalysis

orndorffgrant opened the pull request:
Fixes: #1

Here are some sample commands to try it out:
```
python run.py --feature 1 --user sdispater
python run.py --feature 1 --user abn
python run.py --feature 1 --user radoering
```

Example plot:
<img width="1608" height="924" alt="Screenshot_20251101_213552" src="https://github.com/user-attachments/assets/27dd45bf-0018-4883-87c9-6343fa02da6e" />

abhaumik13 commented on config.json:
```diff
@@ -1,3 +1,3 @@
-    "ENPM611_PROJECT_DATA_PATH":"path/to/data/file.json"
-}
\ No newline at end of file
+    "ENPM611_PROJECT_DATA_PATH":"./downloads/poetry_issues_all.json"
```
We need to ensure before submission that we are not committing local system file path here

orndorffgrant replied:
Done!

abhaumik13 reviewed (dismissed):
Functional Testing - Completed
- Successfully ran the code locally
- Tested with sample data from "poetry_issues_all.json"

Code Review - Completed
- Note to self/group: Need to remove the reference to the local file path in config.json before final commit

Overall Recommendation: **Approve**

aastha2912 commented:
Suggestion: 
1. Would be helpful if you could add screenshots or sample output in the PR itself - makes it easier to quickly see what the feature is doing.

aastha2912 commented:
Testing I did:

1. Ran it with the full issues JSON - works fine.
2. Also tried using --label flag - it runs, but it doesn’t actually filter anything. If this feature doesn't support --label, maybe the script should throw an error or at least print a message, so it’s not confusing.
3. Better to have a label on the Y-axis as well 

Overall: Code works, just a couple of small suggestions. Good to merge after that.

orndorffgrant committed:
feature-1

Fixes: #1

orndorffgrant committed:
address review feedback

orndorffgrant commented:
Thanks @aastha2912 and @abhaumik13!

* Removed config.json changes.
* Added y-axis labels.
* Now raising exception on --label flag.
* Added screenshot to PR description (there is no meaningful output on stdout)

abhaumik13 reviewed (approved):
Looks good! I have tested the functionality, with the various arguments, and the results look as expected. Approving the changes.

aastha2912 commented:
Thanks! Everything working as expected. All comments addressed. Approving
